rules/no-single-promise-in-promise-methods.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+const {isMethodCall} = require('./ast/index.js');
+const needsSemicolon = require('./utils/needs-semicolon.js');
+
+const MESSAGE_ID_ERROR = 'no-single-promise-in-promise-methods/error';
+const MESSAGE_ID_SUGGESTION_1 = 'no-single-promise-in-promise-methods/suggestion-1';
+const MESSAGE_ID_SUGGESTION_2 = 'no-single-promise-in-promise-methods/suggestion-2';
+const messages = {
+	[MESSAGE_ID_ERROR]: 'Parameter in `Promise.{{method}}()` should not be a single element array.',
+	[MESSAGE_ID_SUGGESTION_1]: 'Use the value directly.',
+	[MESSAGE_ID_SUGGESTION_2]: 'Wrap the value in a `Promise.resolve`.',
+};
+const METHODS = ['all', 'any', 'race'];
+
+const isPromiseMethodCallWithSingleElementArray = node =>
+	isMethodCall(node, {
+		object: 'Promise',
+		methods: METHODS,
+		optionalMember: false,
+		optionalCall: false,
+		argumentsLength: 1,
+	})
+	&& node.arguments[0].type === 'ArrayExpression'
+	&& node.arguments[0].elements.length === 1
+	&& node.arguments[0].elements[0] !== null
+	&& node.arguments[0].elements[0].type !== 'SpreadElement';
+
+const getText = ({sourceCode}, node, element, prefix = '', suffix = '') => {
+	const previousToken = sourceCode.getTokenBefore(node);
+	const text = sourceCode.getText(element);
+	const parenthesizedText = element.type === 'SequenceExpression' ? `(${text})` : text;
+	const wrappedText = `${prefix}${parenthesizedText}${suffix}`;
+
+	return needsSemicolon(previousToken, sourceCode, wrappedText) ? `;${wrappedText}` : wrappedText;
+};
```

Comment:
There is an existing `getParenthesizedText` utility we can use directly.

Reply:
It's done.